source/cellular_bg96_api.c

```diff
@@ -2597,9 +2727,15 @@ CellularError_t Cellular_SocketRecv( CellularHandle_t cellularHandle,
          * The max length of the string is fixed and checked offline. */
         /* coverity[misra_c_2012_rule_21_6_violation]. */
         ( void ) snprintf( cmdBuf, CELLULAR_AT_CMD_TYPICAL_MAX_SIZE,
-                           "%s%ld,%ld", "AT+QIRD=", socketHandle->socketId, recvLen );
-        pktStatus = _Cellular_TimeoutAtcmdDataRecvRequestWithCallback( pContext,
-                                                                       atReqSocketRecv, recvTimeout, socketRecvDataPrefix, NULL );
+                           "%s%ld,%ld", (useSsl ?  "AT+QSSLRECV=" : "AT+QIRD="), socketHandle->socketId, recvLen );
+        
+        if(useSsl){        
+            pktStatus = _Cellular_TimeoutAtcmdDataRecvRequestWithCallback( pContext,
+                                                                       atReqSocketRecv, recvTimeout, socketRecvDataPrefixSsl, NULL );
+        }else{
+            pktStatus = _Cellular_TimeoutAtcmdDataRecvRequestWithCallback( pContext,
+                                                                       atReqSocketRecv, recvTimeout, socketRecvDataPrefixNonSsl, NULL );
+        }

```

Comment:
To improve readabilty consider the following:
```
CellularATCommandDataPrefixCallback_t callback;
if (useSsl)
{
  callback = socketRecvDataPrefixSsl;
}
else
{
  callback = socketRecvDataPrefixNonSsl;
}
pktStatus = _Cellular_TimeoutAtcmdDataRecvRequestWithCallback(atReqSocketRecv, recvTimeout, callback, NULL );
```

Reply:
Done